File: ios/AITutor/Data/Remote/SupabaseProvider.swift
import Foundation
import Supabase

/// Single source of truth for the Supabase client. Other layers depend on
/// this provider (or repositories that wrap it) instead of constructing
/// clients themselves.
final class SupabaseProvider {
    static let shared = SupabaseProvider()

    let client: SupabaseClient

    private init() {
        client = SupabaseClient(
            supabaseURL: Secrets.supabaseURL,
            supabaseKey: Secrets.supabaseAnonKey,
            options: SupabaseClientOptions(
                auth: .init(
                    redirectToURL: URL(string: "aitutor://auth-callback"),
                    flowType: .pkce
                )
            )
        )
    }
}

File: android/app/src/main/java/com/aitutor/app/data/remote/SupabaseProvider.kt
package com.aitutor.app.data.remote

import com.aitutor.app.BuildConfig
import io.github.jan.supabase.SupabaseClient
import io.github.jan.supabase.auth.Auth
import io.github.jan.supabase.createSupabaseClient
import io.github.jan.supabase.functions.Functions
import io.github.jan.supabase.postgrest.Postgrest
import io.github.jan.supabase.realtime.Realtime
import io.github.jan.supabase.storage.Storage
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Single source of truth for the Supabase client. Other layers should depend
 * on this provider (or on repositories that wrap it) rather than constructing
 * clients themselves.
 */
@Singleton
class SupabaseProvider @Inject constructor() {

    val client: SupabaseClient by lazy {
        createSupabaseClient(
            supabaseUrl = BuildConfig.SUPABASE_URL,
            supabaseKey = BuildConfig.SUPABASE_ANON_KEY
        ) {
            install(Auth) {
                // Deep-link redirect handler (matches AndroidManifest filter)
                scheme = "aitutor"
                host = "auth-callback"
            }
            install(Postgrest)
            install(Realtime)
            install(Storage)
            install(Functions)
        }
    }
}
